### ISSUE TITLE
So, what is it that determines if a mod shows up in the "Add Mods" window?

### ISSUE THREAD

**Issue by jbrown1597** [killpowa/Twickt-Launcher#1581]
I adore this launcher so much.

One thing I can't wrap my head around though, is why a healthy amount of the mods I search for aren't available to download through GD. Mods that are: 

- From Curseforge/Modrinth
- Work on current versions of MC (1.20.1, what I use)
- Have been around for at least a few months

Any info regarding this would be awesome!

P.S - Will GD update mods that I had to download manually from Curse/Modrinth? It recognizes them immediately, so I am hopeful, but unsure nonetheless.

**Comment by Eskaan** [comment]
I'd assume those mods are ones where the Authors have removed them from the API to make more profit. CurseForge big company money stuff. Can't say much more without mod examples.

No, I don't think the launcher can recognize mods that are downloaded by hand. Especially not if there the ones I talked about above.

**Comment by jbrown1597** [comment]
Thanks for the response!

1. Ahh, figures. Good example would be any mod made by Serilum, none of them show up.
2. Well damn. I'm way too far gone now lol. 

I learned the launcher does indeed recognize manually placed mods IF they're already available to download via GD. What I do wonder is if those ones will play nice with updates.

Anywho,
So essentially, now, in order to determine which mods WON'T auto-update, I need to go through my entire mod list & search one-by-one in GD, noting which mods don't show up in search - those being the culprits. Agree or disagree?

**Comment by Eskaan** [comment]
I think it will show you available updates when you try to click on the mod (opening the mod's page).

**Comment by jbrown1597** [comment]
Lemme try wording it a different way:

1. Mods that GD DOES recognize when installed, AND are downloadable thru GD:   Yes, can auto-update
2. Mods that GD DOESN'T recognize when installed, AND are added manually:   No, can't auto-update
3. Mods that GD DOES recognize when installed, BUT aren't available to download in GD / are added manually:   ???

Can you provide clarity on number 3?

Will all mods that fall under number 3 need to be updated manually?

**Comment by Eskaan** [comment]
I think so, but I am not really sure about it. I am basically just maintaining the issue tracker at this point, so there is no one really working on the project anymore. The core team has moved on to the rewrite.

**Comment by jbrown1597** [comment]
Rewrite, as in you're essentially remaking the program from scratch?

While I don't take joy in saying this, I fear it's too little too late - Modrinth Launcher just released a few weeks ago & its' already miles ahead of GDLauncher. 

Funny enough, I had the same issue using chat via Modrinth Launcher. One quick discord message later & chat was fixed the very next day.

Regardless, I sincerely do hope you / your team finds success <3
Thanks for the replies!

**Comment by Eskaan** [comment]
Yes, there is a rewrite being made from scratch and there is already a (quite buggy) version available in alpha.

I myself also prefer the modrinth launcher tho.
